code/modules/mob/living/living.dm

```diff
@@ -92,6 +92,8 @@
 		FireBurn(11, 9001, ONE_ATMOSPHERE) // lag free weird way of doing it
 		fire_stacks = 11
 		ignite() // ffffFIRE!!!! FIRE!!! FIRE!!
+	else if(istype(get_turf(src),/turf/unsimulated/floor/planetary/toxic))
```

Comment:
better to call some modular proc, for example

`turf/proc/onLife(mob/living/L)`